src/profile-logic/marker-data.js

```diff
@@ -243,10 +255,108 @@ export function extractMarkerDataFromName(
   return newMarkers;
 }
 
+/**
+ * This class is a specialized Map for IPC marker data, using the thread ID and
+ * marker index as the key, and storing IPCPairData (which holds the start/end
+ * times for the IPC message.
+ */
+export class IPCMarkerCorrelations {
+  _correlations: Map<string, IPCPairData>;
+
+  constructor() {
+    this._correlations = new Map();
+  }
+
+  _makeKey(tid: number | void, index: number) {
+    if (tid === undefined) {
+      return index.toString();
+    }
+    return `${tid},${index}`;
+  }
+
+  set(tid: number | void, index: number, data: IPCPairData): void {
+    this._correlations.set(this._makeKey(tid, index), data);
+  }
+
+  get(tid: number | void, index: number): ?IPCPairData {
+    return this._correlations.get(this._makeKey(tid, index));
+  }
+}
+
+/**
+ * This function correlates the sender and recipient sides of IPC markers so
+ * that we can share data between the two during profile processing.
+ */
+export function correlateIPCMarkers(threads: Thread[]): IPCMarkerCorrelations {
+  // Create a unique ID constructed from the source PID, destination PID,
```

Comment:
I would check `stringTable.hasString('IPC')` here, so we can bail out early if there are no IPC markers. It could be wrong, but I tested a few place and it was a pretty good "cheap" check.

Reply:
You handled this while i was reviewing :)